var/spack/repos/builtin/packages/sirius/package.py

```diff
@@ -80,17 +85,19 @@ class Sirius(CMakePackage, CudaPackage):
     depends_on('py-voluptuous', when='+python', type=('build', 'run'))
     depends_on('py-pybind11', when='+python', type=('build', 'run'))
     depends_on('magma', when='+magma')
+    depends_on('boost cxxstd=14 +filesystem \
+                ~atomic ~chrono ~date_time ~exception \
+                ~graph ~iostreams ~locale ~log ~math ~program_options \
+                ~random ~regex ~serialization ~signals ~system ~test \
+                ~thread ~timer ~wave', when='+boost_filesystem')
```

Comment:
wouldn't it be sufficient to depend on `boost cxxstd=14 +filesystem`? A boost which has also the `regex` should still work, right?

Reply:
Yeah, that's most composable, but also pulling in a lot of stuff I don't need :(

Maybe far-fetched but if another package depends both on `boost+math` and `sirius+boost_filesystem` then this might result in a concretization error (`+math` and `~math` required). I'll just simplify this to `boost cxxstd=14 +filesystem`, but it's a bit sad :(. Maybe we can disable all variants in boost by default.

Reply:
well, we had the same issue in Gentoo Linux with the USE flags: you got to find a sane subset of flags to enable by default (following upstream default when possible, or maybe what other Linux distros enable).